controllers/config/clientconfig_controller_test.go

```diff
@@ -0,0 +1,200 @@
+package config
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/clientcache"
+	testutils "github.com/k8ssandra/k8ssandra-operator/pkg/test"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/utils"
+	"github.com/k8ssandra/k8ssandra-operator/test/framework"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+)
+
+const (
+	timeout  = time.Second * 5
+	interval = time.Millisecond * 500
+)
+
+var (
+	testEnv     *testutils.MultiClusterTestEnv
+	scheme      *runtime.Scheme
+	logger      logr.Logger
+	cancelCalls int
+	// secretFilter map[types.NamespacedName]types.NamespacedName
+	reconciler *ClientConfigReconciler
+	usedMgr    manager.Manager
+	kubeConfig []byte
+)
+
+func TestClientConfigReconciler(t *testing.T) {
```

Comment:
Can you add or update tests to verify shutdown happens when:

* ClientConfig is deleted
* ClientConfig or secret is missing hash annotations
* Hash annotations have changed